docs/tofu.logging.recipes.zio.md

```diff
@@ -0,0 +1,168 @@
+whenever we want to specify the dependency on a service.
+3. Logging methods can be used:
+   * explicitly like `log.info("Start getting datetime")`
+   * via string interpolation provided by [tofu.syntax.logging](./tofu.logging.syntax.md). For this purpose 
+   log service was defined as `implicit`.
+4. To log object Tofu must know how to present it in the log message. This way is described by instances of `Loggable`
+type class. We did provide no one because Tofu already has `java.time` `Loggable` instances.
+
+Now look at the main app:
+```scala
+import tofu.logging.zlogs.ZLogging
+import zio._
+import zio.magic._
+
+object Main extends zio.App {
+  override def run(args: List[String]): URIO[zio.ZEnv, ExitCode] = {
+    for {
+      barService <- ZIO.service[BarService]
+      _ <- barService.foo
+    } yield ()
+  }.injectCustom(
+    BarService.live,
+    ZLogging.Make.layerPlain
+  ).exitCode
+}
+```
+Thanks to [zio-magic](https://github.com/kitlangton/zio-magic/) we can just list all dependencies in 
+the `injectCustom` method, not design layers manually. Output of the program will be:
+```json lines
+{"@timestamp":"2021-09-19T21:12:17.746Z","loggerName":"BarServiceImpl","threadName":"zio-default-async-4","level":"INFO","message":"Start program"}
+{"@timestamp":"2021-09-19T21:12:18.017Z","loggerName":"BarServiceImpl","threadName":"zio-default-async-4","level":"DEBUG","message":"Got current date 2021-09-20"}
+```
+### Custom `Loggable`
+Default `Loggable[LocalDate]` is based on `stringValue` instance, hence the date was logged as a plain string, not key-value. 
+What if you want the date to be a separated field in json? Well, you can customize `Loggable`. Add the following line into the `BarServiceImpl`:
+```scala
+private implicit val customDateLoggable = Loggable[LocalDate].named(name="date")
+```
+Now the output looks like:
+```json lines
+{"@timestamp":"2021-09-19T21:12:18.017Z","loggerName":"BarServiceImpl","threadName":"zio-default-async-4","level":"DEBUG","message":"Got current date 2021-09-20","date":"2021-09-20"}
+```
+`Loggable.apply[LocalDate]` summons an instance from the global scope, `.named` converts logged object into a single field named `name`.
+There are several methods to create and modify `Loggable` instances, read more in [Loggable](./tofu.logging.loggable.md) section.
+
+### Context logging
+Let's consider how to log the context along with actual log message. If you have an instance of `Loggable` for your context, 
+you can have it logged automagically. For example:
+```scala
+import derevo.derive
+import tofu.logging.derivation.loggable
+
+@derive(loggable)
+final case class Context(requestId: Int)
+```
+Here we use [Tofu Derevo](https://github.com/tofu-tf/derevo) for automatic derivation `Loggable` instance.
+
+The main idea of this approach is to store your in ZIO [FiberRef](https://zio.dev/docs/datatypes/fiber/fiberref). 
```

Comment:
What if someone would want to store the Context not in FiberRef, but in R itself (which is blasphemy of course)? Should we have a note about this approach here?

Reply:
Yes, we should. But I've never use the R as a context. It may be useful with TF and WithContext monad, but I'm not good in TF, so I'm not able to write a good example with that approach.

Now I can note that this is one of possible ways